synapse/lib/view.py

```diff
@@ -125,11 +130,259 @@ async def __anit__(self, core, node):
         # isolate some initialization to easily override.
         await self._initViewLayers()
 
+        self.readonly = self.wlyr.readonly
+
         self.trigtask = None
         await self.initTrigTask()
 
         self.mergetask = None
 
+        self.tagcache = s_cache.FixedCache(self._getTagNode, size=self.tagcachesize)
+
+        self.nodecache = collections.deque(maxlen=self.nodecachesize)
+        self.livenodes = weakref.WeakValueDictionary()
+
+    def clearCache(self):
+        self.tagcache.clear()
+        self.nodecache.clear()
+        self.livenodes.clear()
+
+    def clearCachedNode(self, nid):
+        self.livenodes.pop(nid, None)
+
+    async def saveNodeEdits(self, edits, meta, bus=None):
+        '''
+        Save node edits and run triggers/callbacks.
+        '''
+
+        if self.readonly:
+            mesg = 'The view is in read-only mode.'
+            raise s_exc.IsReadOnly(mesg=mesg)
+
+        callbacks = []
+
+        wlyr = self.layers[0]
+
+        # hold a reference to  all the nodes about to be edited...
+        nodes = {e[0]: await self.getNodeByNid(e[0]) for e in edits if e[0] is not None}
+
+        saveoff, nodeedits = await wlyr.saveNodeEdits(edits, meta)
+
+        fireedit = (bus is not None and bus.viewiden == self.iden)
```

Comment:
This does have the effect of firing the edits caused `view.exec` with the current view.